src/main/java/com/minecolonies/coremod/entity/ai/util/Structure.java

```diff
@@ -80,14 +110,47 @@ public SchematicBlock(final Block block, final BlockPos blockPosition, final IBl
             this.worldBlock = worldBlock;
             this.worldMetadata = worldMetadata;
         }
+
+        /**
+         * Checks if the structureBlock equals the worldBlock.
+         * @return true if so.
+         */
+        public boolean doesStructureBlockEqualWorldBlock()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "doesStructureBlockEqualWorldBlock" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)